cmake/FindZ3.cmake

```diff
@@ -35,17 +35,34 @@ find_program(Z3_EXEC
 
 mark_as_advanced(Z3_FOUND Z3_INCLUDE_DIR Z3_LIBRARY Z3_EXEC)
 
-include(FindPackageHandleStandardArgs)
-find_package_handle_standard_args(Z3 DEFAULT_MSG 
-  Z3_INCLUDE_DIR 
-  Z3_LIBRARY
-)
+if("${CMAKE_SYSTEM_NAME}" MATCHES "[Ww]indows")
 
-if(Z3_FOUND)
+  find_package(Z3
+    REQUIRED
+    CONFIG
+    # `NO_DEFAULT_PATH` is set so that -DZ3_DIR has to be passed to find Z3.
+    # This should prevent us from accidentally picking up an installed
+    # copy of Z3. This is here to benefit Z3's build system when building
+    # this project. When making your own project you probably shouldn't
+    # use this option.
+    NO_DEFAULT_PATH
+  )
 
-  set(Z3_INCLUDE_DIRS ${Z3_INCLUDE_DIR})
+  set(Z3_INCLUDE_DIR ${Z3_CXX_INCLUDE_DIRS})
+  set(Z3_LIBRARY ${Z3_LIBRARIES})
+  set(Z3_VERSION ${Z3_VERSION_STRING})
 
-  set(Z3_LIBRARIES ${Z3_LIBRARY})
+else()
+
+  include(FindPackageHandleStandardArgs)
```

Comment:
Try changing the path prefix or hint. This should work on Windows if proper path and hints are provided.